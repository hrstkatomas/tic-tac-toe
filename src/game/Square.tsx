import React, { useState } from 'react'

type reactHook = [
  string,
  (value: string) => void
]

const Square: React.FunctionComponent = () => {
  const [ value, setValue ]: reactHook = useState('')

  return (
    <button onClick={() => setValue('x')}>
      {value}
    </button>
  )
}

export default Square
