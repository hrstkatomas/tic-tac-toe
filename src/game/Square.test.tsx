import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Square from './Square'

describe("Board component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Square />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders according to snapshot', () => {
    const tree = renderer
      .create(<Square />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
