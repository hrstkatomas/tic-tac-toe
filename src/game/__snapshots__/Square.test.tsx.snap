// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component renders according to snapshot 1`] = `
<button>
  x
</button>
`;
